### ISSUE TITLE
restore terminal when an exception occurs

### ISSUE THREAD

**Issue by TomasTomecek** [TomasTomecek/sen#68]
http://stackoverflow.com/questions/2649733/hide-cursor-on-remote-terminal/2676973

**Comment by ibizaman** [comment]
Ah yes, this would be very useful since they appear quite often.
But then we loose the stacktrace I suppose. Could the stacktrace be saved to a log file somewhere?

**Comment by TomasTomecek** [comment]
[git master](https://github.com/TomasTomecek/sen/blob/2fad2ab7bd4dc5bf07876a87348c216706e47bf4/sen/cli.py#L78) should contain some fixes related to this -- I'm planning to improve situation in following weeks.

**Comment by ibizaman** [comment]
Awesome, I'll use that and report the one I have left. Thanks for the awesome work on this awesome tool!

**Comment by TomasTomecek** [comment]
Yep, that's suppose to be an essential feature of sen. Unfortunately the tui framework doesn't support that and I need to do some hardcore voodoo to make it work properly (without tracebacks, lockups and glitches)

**Comment by TomasTomecek** [comment]
It seems like there's a fix for this in urwid 1.3.1:
```
Fix for screen not getting reset on exception regression (by Rian Hunter)
```
http://urwid.org/changelog.html#urwid-1-3-1

I can confirm that upon exception, my terminal is restored. Closing then.
